Add multi-select component using multiselect-react-dropdown

RyanPrice-King opened the pull request.

dsternlicht commented on src/components/formRow/formRow.comp.tsx:
```diff
@@ -1,6 +1,7 @@
 import React, { useState } from 'react';
 import { orderBy } from 'natural-orderby';
 import { toast } from 'react-toastify';
+import { Multiselect } from 'multiselect-react-dropdown';
```
Why not using a more popular (and maintained) library such as `react-select`?
https://react-select.com/home

RyanPrice-King replied:
I had problems with others importing the `finalOptions` object and found this one to contain additional parameters that were useful like the `selectionLimit` for multi.  Also, it allowed me to use the same component for single and multi-select.

dsternlicht commented on src/components/formRow/formRow.comp.tsx:
```diff
@@ -172,19 +174,34 @@ export const FormRow = withAppContext(({ context, field, direction, showReset, o
+          };
+
           return (
-            <select {...inputProps()}>
```
For non multi select, maybe we should use a simple selectbox?

RyanPrice-King replied:
I did originally, but seemed redundant to me as I could use the same component for both where multiple lines were originally duplicated across both components and this reduced the lines of code needed and simplified the code.  Also, with this approach I felt it is better for the end-user to not have to think about other types and to just add a simple multi switch to the select type.

dsternlicht commented:
@RyanPrice-King this is how it displays right now after I pulled the changes:
![Screen Shot 2021-12-14 at 8 38 27](https://user-images.githubusercontent.com/1046545/145946266-950cb67e-6e3f-4f8e-b0fe-5bb4caa7c305.png)
![Screen Shot 2021-12-14 at 8 38 34](https://user-images.githubusercontent.com/1046545/145946277-c30b3f67-cee5-41b7-b17b-1e329aae7054.png)

dsternlicht reviewed (changes requested):
Something is wrong with the styles:
https://github.com/dsternlicht/RESTool/pull/193#issuecomment-993207308

RyanPrice-King commented:
![image](https://user-images.githubusercontent.com/34060058/145970450-043b6f6c-3551-4ccf-8b64-cadd27fd3b37.png)
It works with optionSource.

I will need to find out why it is not working with config options I think.

dsternlicht commented:
> ![image](https://user-images.githubusercontent.com/34060058/145970450-043b6f6c-3551-4ccf-8b64-cadd27fd3b37.png) It works with optionSource.
> 
> I will need to find out why it is not working with config options I think.

Ok. For non-multi source select boxes, let's simply use the regular selectbox (as we had until now).

RyanPrice-King commented:
Ok, I have now found the problem and fixed it and added an object check like what you had before as that was one of the issues here, but if you prefer to split it still I will do that as well. and push the changes.

RyanPrice-King commented:
Either way, can I ask you not to merge yet please as I am testing for other improvements as I build the site I require and may make further improvements.

dsternlicht commented:
@RyanPrice-King no problem, I'm not merging yet. For the non-multi options, let's use the old selectbox.

RyanPrice-King commented:
Yes, will do :)

RyanPrice-King commented:
I have separated the components like you asked and noticed the select-multi was not loading pre-selected values in the dropdown, so I have fixed that now too.  Still testing and will let you know when I am happy with it.
